pkg/metrics/sampler/matcher.go

```diff
@@ -204,3 +213,52 @@ func (ne constantMatcher) Evaluate(_ interface{}) bool {
 func (ne constantMatcher) String() string {
 	return fmt.Sprint(ne.value)
 }
+
+// NewSampleMatchFn creates new includeSampleMatchFn func, enableProcessMetrics might be nil when
+// value was not set.
+func NewSampleMatchFn(enableProcessMetrics *bool, includeMetricsMatchers config.IncludeMetricsMap, ffRetriever feature_flags.Retriever) IncludeSampleMatchFn {
+	// configuration option always takes precedence over FF and matchers configuration
+	if enableProcessMetrics != nil {
+		if *enableProcessMetrics == false {
+			trace.MetricMatch("EnableProcessMetrics is FALSE, process metrics will be DISABLED")
+			return func(sample interface{}) bool {
+				// no process samples are included
+				_, ok := sample.(types.ProcessSample)
+				return !ok
+			}
+		} else {
+			ec := NewMatcherChain(includeMetricsMatchers)
+			if ec.Enabled {
+				trace.MetricMatch("EnableProcessMetrics is TRUE and rules ARE defined, process metrics will be ENABLED for matching processes")
+				return func(sample interface{}) bool {
+					return ec.Evaluate(sample)
+				}
+			}
+
+			trace.MetricMatch("EnableProcessMetrics is TRUE and rules are NOT defined, ALL process metrics will be ENABLED")
+			return func(sample interface{}) bool {
+				// all process samples are included
+				return true
+			}
+		}
+	}
+
+	// if config option is not set, check if we have rules defined. those take precedence over the FF
+	ec := NewMatcherChain(includeMetricsMatchers)
+	if ec.Enabled {
+		trace.MetricMatch("EnableProcessMetrics is EMPTY and rules ARE defined, process metrics will be ENABLED for matching processes")
+		return func(sample interface{}) bool {
+			return ec.Evaluate(sample)
+		}
+	}
+
+	// configuration option is not defined and feature flag is present, FF determines, otherwise
+	// all process samples will be excluded
+	return func(sample interface{}) bool {
+		if enabled, exists := ffRetriever.GetFeatureFlag(handler.FlagFullProcess); exists {
+			return enabled
+		}
+		_, ok := sample.(types.ProcessSample)
```

Comment:
this assertion needs to wrap the FF check. We should check the FF only if the sample is a ProcessSample, otherwise return true (for the other types of samples which are always collected)